app/models/grouping.rb

```diff
@@ -695,6 +695,17 @@ def last_commit_date
     end
   end
 
+  # Helper for populate_submission_table
+  # Returns boolean value based on if the submission has files or not
+  def has_files_in_submission?
```

Comment:
Return `!has_submission || SubmissionFile.where(...).exists?` instead of using a conditional.